src/main/java/com/main/service/impl/FollowerSerivceImpl.java

```diff
@@ -0,0 +1,99 @@
+package com.main.service.impl;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import com.main.dto.FollowerDto;
+import com.main.dto.ResponseDto;
+import com.main.entity.Follower;
+import com.main.entity.Person;
+import com.main.repository.FollowerRepository;
+import com.main.repository.PersonRepository;
+import com.main.service.FollowerService;
+
+@Service
+public class FollowerSerivceImpl implements FollowerService {
+
+	private FollowerRepository followerRepository;
+
+	private PersonRepository personRepository;
+
+	public FollowerSerivceImpl(FollowerRepository followerRepository, PersonRepository personRepository) {
+		this.followerRepository = followerRepository;
+		this.personRepository = personRepository;
+	}
+
+	public ResponseDto<String> allFollowers(String user) {
+
+		ResponseDto<String> response = new ResponseDto<>();
+
+		// First, find the person by their ID
+		Person person = personRepository.findByName(user).orElse(null);
+		// If the person is found, retrieve all messages from that person
+		if (person != null) {
```

Comment:
I think this kind of `if` statements are difficult to read. For the case of the service is better to use what is called `guard blocks`. You first check that all the conditions you need are satisfied and then you continue. For example this could be done as:

```java
if (person == null) {
  response.setStatus(404);
  response.setMessage("No user found");
  response.setData(Collections.emptyList());
  return response
}
// Then check result and set response
...
``` 

Instead of using the if/else that sometimes are hard to read.

Reply:
I will change this, to make easy to uderstand